### ISSUE TITLE
kendo ui DataSource error handler parameters

### ISSUE THREAD

**Issue by dimokdimon** [jaystack/jaydata#92]
Here is your code that creates fail callback.
      destroy: function (options, model) {
        var query = self;
        query.entityContext.onReady().then(function () {
          if (model.length > 1) {
            model.forEach(function (item) {
              ctx.remove(item.innerInstance());
            });
            ctx.saveChanges().then(function () {
              options.success({ data: options.data });
            })<b>.fail(function () {</b>
              ctx.stateManager.reset();
              //alert("error in save:" + arguments[0]);
              <b>options.error({}, "error", options.data);</b>
            });
          } else {
            model[0].innerInstance().remove().then(function () {
              options.success({ data: options.data });
            })<b>.fail(function () {</b>
              ctx.stateManager.reset();
              //alert("error in save:" + arguments[0]);
              <b>options.error({}, "error", options.data);</b>
            });
          }
        });
When <b>option.error</b> is called you pass <b>{}</b> as first parameter, meaning that error handler does not receive the original <b>xhr</b>.  I've changed <b>.fail(function(){</b> to <b>.fail(function (exception) {</b> and <b>options.error({}, "error", options.data);</b> to <b>options.error(exception, "error", options.data);</b>  . It work fine now (with kendo ui). Is there any problem with this approach?

**Comment by agdolla** [comment]
Hi
It should be ok. I remember doing that part and I could not find a good documentation about error handling with kendoui datasource. If you've found a good doc about it please share it with us. Thanks

**Comment by dimokdimon** [comment]
I don’t think it’s important what kendo’s handler is sending to callback.
Whatever it is - it’s kendo-specific  and kendo users most likely will be able to make a sense out of it.
Your goal should be just pass it through to client’s callback. The rest is client’s responsibility.
And this is probably true for other providers.

Question is – will you treat your current approach as a bug and implement the idea behind my fix or I’ll have to manually change your code when I upgrade to new version?

**Comment by jaydata** [comment]
Yeah. I agree that our implementation needs to be changed. Can you just create a pull request? Your efforts are greatly appreciated by us and by the community

dimokdimon notifications@github.com wrote:

I don’t think it’s important what kendo’s handler is sending to callback.
Whatever it is - it’s kendo-specific and kendo users most likely will be able to make a sense out of it.
Your goal should be just pass it through to client’s callback. The rest is client’s responsibility.
And this is probably true for other providers.

Question is – will you treat your current approach as a bug and implement the idea behind my fix or I’ll have to manually change your code when I upgrade to new version?

**Comment by dimokdimon** [comment]
I am not sure what you mean by "create a pull request".- this is my first ever interaction of that kind.
Do you want me to pull the source code, apply the the change and push it back?

**Comment by robesz** [comment]
This issue needs a further investigation, we update this thread with results and collaborate.
